sui-execution/latest/sui-move-natives/src/object_runtime/mod.rs

```diff
@@ -573,3 +604,50 @@ fn update_owner_map(
     }
     Ok(())
 }
+
+// TODO use a custom DeserializerSeed and improve this performance
+pub fn get_all_uids(
+    fully_annotated_layout: &MoveTypeLayout,
+    bcs_bytes: &[u8],
+) -> Result<BTreeSet<ObjectID>, /* invariant violation */ String> {
+    let mut ids = BTreeSet::new();
+    let v = MoveValue::simple_deserialize(bcs_bytes, fully_annotated_layout)
+        .map_err(|e| format!("Failed to deserialize. {e:?}"))?;
+    get_all_uids_in_value(&mut ids, &v)?;
+    Ok(ids)
+}
+
+fn get_all_uids_in_value(
+    acc: &mut BTreeSet<ObjectID>,
+    v: &MoveValue,
+) -> Result<(), /* invariant violation */ String> {
```

Comment:
To me this is the particularly gross section of this PR, as we have to deserialize the object into `MoveValue` just to traverse to find the UIDs. 

A more optimal approach would be to just walk the bytes `&[u8]` with the `fully_annotated_layout` to look for UIDs, but never actually allocate into another `MoveValue`